activiti-core/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeImpl.java

```diff
@@ -441,6 +447,22 @@ public void save(SaveTaskPayload saveTaskPayload) {
     }
 
 
+    @Override
+    public Task assign(AssignTaskPayload assignTaskPayload) {
+        String assignee = assignTaskPayload.getAssignee();
+        String taskId = assignTaskPayload.getTaskId();
+
+        checkUserHasAccessToTask(taskId);
+        checkAssigneeIsACandidateUser(taskId, assignee);
+        
+        //We need to release, claim for assigned task is not working!
+        taskService.unclaim(taskId);
+        //Now assign a new user
```

Comment:
in my opinion, it would be better to have a releaseAndClaim method and remove the comments